device_doctor/lib/src/process_helper.dart

```diff
@@ -67,3 +61,56 @@ Future<String> eval(String executable, List<String> arguments,
 
   return output.trimRight();
 }
+
+/// Splits [from] into lines and selects those that contain [pattern].
+Iterable<String> grep(Pattern pattern, {@required String from}) {
+  return from.split('\n').where((String line) {
+    return line.contains(pattern);
+  });
+}
+
+Future<bool> killAllRunningProcessesOnWindows(String processName, {ProcessManager processManager}) async {
+  processManager ??= LocalProcessManager();
+  // Avoid endless loop when a process from a different use exists, and fails
+  // to get killed every try.
+  int tries = 3;
+  while (tries > 0) {
+    tries--;
+    final pids = runningProcessesOnWindows(processName, processManager: processManager);
+    if (pids.isEmpty) {
+      return true;
+    }
+    for (String pid in pids) {
+      processManager.runSync(<String>['taskkill', '/pid', pid, '/f']);
+    }
+    // Killed processes don't release resources instantenously.
+    const Duration delay = Duration(seconds: 1);
+    await Future<void>.delayed(delay);
+  }
+  return false;
+}
+
+List<String> runningProcessesOnWindows(String processName, {ProcessManager processManager}) {
```

Comment:
Same as above, different file?

Reply:
Also do we really need this from a device_doctor tool? or should the tool be renamed to something else to account for host related functionality?

Reply:
These two functions are used to serve the device health check. They are needed when we first fail to list devices and need a retry
I believe these logics have been added to cover some corner cases.